File: Task-03/C/Sub-Task 3/Pattern.c
#include <stdio.h>

int main() {
    int n, i, j;
    printf("Enter a number: ");
    scanf("%d", &n);

    for (i = 0; i < n; i++) {
        for (j = 0; j < n - i - 1; j++) printf(" ");
        for (j = 0; j < 2 * i + 1; j++) printf("*");
        printf("\n");
    }
    for (i = n - 2; i >= 0; i--) {
        for (j = 0; j < n - i - 1; j++) printf(" ");
        for (j = 0; j < 2 * i + 1; j++) printf("*");
        printf("\n");
    }

    return 0;
}

File: Task-03/Haskell/Sub-Task 3/Pattern.hs
main = do
    putStrLn "Enter a number: "
    input <- getLine
    let n = read input :: Int
    putStr $ unlines $ diamond n

diamond :: Int -> [String]
diamond n = upper ++ lower
    where
        upper = [replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- [0..n-1]]
        lower = [replicate (i + 1) ' ' ++ replicate (2 * (n - i - 2) + 1) '*' | i <- [0..n-2]]
